snorkel/classification/training/loggers/checkpointer.py

```diff
@@ -2,54 +2,82 @@
 import logging
 import os
 from shutil import copyfile
-from typing import Any, Dict, List, Set
+from typing import Any, Dict, Iterable, List, Optional, Set
 
 from snorkel.classification.snorkel_classifier import SnorkelClassifier
-from snorkel.classification.snorkel_config import default_config
-from snorkel.classification.utils import recursive_merge_dicts
+from snorkel.types import Config
 
 Metrics = Dict[str, float]
 
 
+class CheckpointerConfig(Config):
+    """Manager for checkpointing model.
+
+    Parameters
+    ----------
+    checkpoint_dir
+        The path to a directory where checkpoints will be saved
+        The Trainer will set this to the log directory if it is None
+    checkpoint_factor
+        Check for a best model every this many evaluations. For example, if
+        evaluation_freq is 0.5 epochs and checkpoint_factor is 2, then checkpointing
+        will be attempted every 1 epochs.
+    checkpoint_metric
+        The metric to checkpoint on, of the form "task/dataset/split/metric:mode" where
+        mode is "min" or "max".
+    checkpoint_task_metrics
+        Additional metrics to save best models for. Note that the best model according
+        to `checkpoint_metric` will be the one that is loaded after training and used
+        for early stopping.
+    checkpoint_runway
+        No checkpointing will occur for the first this many checkpoint_units
+    checkpoint_clear
```

Comment:
could be misleading naming; maybe `checkpoint_clear_except_best`? or something less verbose...

Reply:
I think the usage of checkpoint_clear is fuzzy at best. It only matters in cases where you want to checkpoint on multiple metrics but you don't intend to ever use the checkpoints saved on the other metrics (if you just care about tracking their progress, you can do that separately from the checkpointer). Added a note to the tracker to remove this option, but saving it for a separate PR, since it's unrelated to the config.